modules/infra/control/port.c

```diff
@@ -626,6 +609,94 @@ static void port_to_api(void *info, const struct iface *iface) {
 	}
 }
 
+static struct event *link_event;
+
+static void link_event_cb(evutil_socket_t, short, void *) {
+	unsigned max_sleep_us, rx_buffer_us;
+	struct rte_eth_rxq_info qinfo;
+	struct rte_eth_link link;
+	struct queue_map *qmap;
+	struct worker *worker;
+	const struct iface *i;
+	struct iface *iface;
+
+	STAILQ_FOREACH (worker, &workers, next) {
+		if (gr_args()->poll_mode)
+			max_sleep_us = 0;
+		else
+			max_sleep_us = 1000; // unreasonably long maximum (1ms)
+
+		arrforeach (qmap, worker->rxqs) {
+			i = port_ifaces[qmap->port_id];
+			if (i == NULL)
+				continue;
+			iface = iface_from_id(i->id);
+			if (iface == NULL)
+				continue;
+
+			if (rte_eth_link_get(qmap->port_id, &link) < 0) {
```

Comment:
Looping on all workers then checking each port means that we check link status multiple times for a single port, that is not optimal.

But I see another potential issue.
Even though a driver reports LSC support, it does not mean that link status is stable between LSC events.
It could result in rare / inconsistent cases where a rxq has a max_sleep_us (port was up at the time), while another rxq does not have one (port was down at the time).

Besides, for ports that don't support LSC, when will the link status change (and speed evaluation / max_sleep evaluation) happen? It may never happen afaiu.